packages-common/pluggable-widgets-generator/generators/app/lib/utils.js

```diff
@@ -0,0 +1,104 @@
+const { join } = require("path");
+const { access, constants, readdir } = require("fs").promises;
+
+function getWidgetDetails(answers) {
+    const widget = Object.create(answers);
+
+    Object.defineProperties(widget, {
```

Comment:
I think you can achieve this simpler by doing `return { ...answers, get name() { ... }, };`

Reply:
I even did it without creating a new object, but just do:
```
Object.defineProperties(answers, {
        name: {
            get() {}
        }...
}
```

Reply:
I think `Object.defineProperties` is a bit too heavy constract

Reply:
I figured out we need to keep the prototyping in place, because the answers object contains identical properties (e.g. `name`). So destructuring answers and defining getters will lead to an infinite loop, since we need to manipulate `name` from answers in the getter. Therefore, we cannot get around `defineProperties`, because setting the prototype after object creation isn't recommended: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/setPrototypeOf

Reply:
Why do we have to preserve a prototype?

Reply:
Changed it